airbyte-integrations/connectors/source-mssql/src/test-integration/java/io/airbyte/integrations/source/mssql/MssqlSourceDatatypeTest.java

```diff
@@ -225,7 +225,7 @@ protected void initTests() {
             .sourceType("time")
             .airbyteType(JsonSchemaType.STRING)
             .addInsertValues("null", "'13:00:01'", "'13:00:04Z'", "'13:00:04.123456Z'")
-            .addExpectedValues(null, "13:00:01.0000000", "13:00:04.0000000", "13:00:04.1234560")
+            .addExpectedValues(null, "13:00:01.000000", "13:00:04.000000", "13:00:04.123456")
```

Comment:
same here - is it the correct behavior to expect milliseconds when inserting data w/o milliseconds?

Reply:
I thought this was your concern in the comment above https://github.com/airbytehq/airbyte/pull/14449#pullrequestreview-1030820583
Previously we had millis in any case. I removed it for cases when source didn't have it.  I thought you had a concern about it so I added it back, but aligned it to the format that we have for other connectors. If it's not OK, kindly ask to specify your expected behavior. Thanks

Reply:
This was a misunderstanding. I wasn't expressing a concern, but asking for the underlying logic/reason for removing trailing zeros in some of the test cases. Generally speaking, we should not be changing source data. If source has time w/o millis we should not be adding them, if source has time with millis, we should not be removing them.